jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/CreateOrUpdateCredentialProfilesAction.kt

```diff
@@ -0,0 +1,97 @@
+// Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.core.credentials
+
+import com.intellij.icons.AllIcons
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.actionSystem.PlatformDataKeys
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.fileEditor.FileEditorManager
+import com.intellij.openapi.fileEditor.OpenFileDescriptor
+import com.intellij.openapi.fileTypes.FileTypes
+import com.intellij.openapi.fileTypes.ex.FileTypeManagerEx
+import com.intellij.openapi.project.DumbAware
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.Messages
+import com.intellij.openapi.vfs.LocalFileSystem
+import org.jetbrains.annotations.TestOnly
+import software.amazon.awssdk.profiles.ProfileFileLocation
+import software.aws.toolkits.core.utils.exists
+import software.aws.toolkits.resources.message
+import software.aws.toolkits.telemetry.AwsTelemetry
+import java.nio.file.Path
+
+class CreateOrUpdateCredentialProfilesAction @TestOnly constructor(
+    private val writer: ConfigFilesFacade
+) : AnAction(message("configure.toolkit.upsert_credentials.action")), DumbAware {
```

Comment:
nit: DumbAwareAction?

Reply:
this file is mostly copy paste from the previous implementation